source/module_esolver/esolver_ks_lcaopw.h

```diff
@@ -19,10 +23,21 @@ namespace ModuleESolver
         /// All the other interfaces except this one are the same as ESolver_KS_PW.
         virtual void hamilt2density(const int istep, const int iter, const double ethr) override;
 
+        void before_all_runners(Input& inp, UnitCell& cell) override;
+        void iter_init(const int istep, const int iter) override;
+        void after_all_runners()override;
+
     protected:
 
         virtual void allocate_hsolver() override;
         virtual void deallocate_hsolver() override;
+        virtual void allocate_hamilt() override;
+
+#ifdef __EXX
```

Comment:
In future, esolver will not contain any class variables.

Reply:
So where to put these objects, global?